### PR TITLE
Feature 03

### PR DESCRIPTION
**Pull request by Sonia-064**

### CODE REVIEW

File: get_request.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+
+@app.route('/get', methods=['GET'])
+def get_request():
+    name = request.args.get('name', 'Guest')
```
Comment (siddharthverma-1607) [inline get_request.py:7]: What does this function do?

File: get_request.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    return jsonify(message=f"Hello, {name}! This is a GET request.")
```
Comment (siddharthverma-1607) [inline get_request.py:8]: What is jsonify?

File: post_sum_numbers.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+@app.route('/post_sum_n', methods=['POST'])
+def post_sum_n():
+    data = request.get_json()
+    numbers = data.get('numbers')
+
+    if not numbers or not isinstance(numbers, list):
+        return jsonify(message="Please provide a list of numbers"), 400
+
+    result = sum(numbers)
```
Comment (siddharthverma-1607) [inline post_sum_numbers.py:13]: what is the data type of numbers?

File: get_request.py (hunk 1 above)
Comment (Sonia-064) [reply get_request.py:7]: The `if __name__ == '__main__':` line checks if the script is run directly (not imported as a module). If it is, it executes `app.run(debug=True)`, which starts the Flask application with debug mode enabled, allowing for live debugging and reloading during development.

File: get_request.py (hunk 2 above)
Comment (Sonia-064) [reply get_request.py:8]: `jsonify` is a function provided by Flask that converts Python data structures (like dictionaries, lists, etc.) into JSON format. It sets the appropriate content-type headers (`application/json`) in the HTTP response.

File: post_sum_numbers.py (hunk 3 above)
Comment (Sonia-064) [reply post_sum_numbers.py:13]: The data types of `num1` and `num2` are not explicitly defined and depend on the JSON payload received; they could be integers, floats, or other types if not validated further.

**Pull request merged**